src/main/java/de/qabel/desktop/ui/actionlog/ActionlogController.java

```diff
@@ -21,75 +31,143 @@
 
 public class ActionlogController extends AbstractController implements Initializable {
 
-
-	ResourceBundle resourceBundle;
 	List<ActionlogItemView> messageView = new LinkedList<>();
 
-
 	@FXML
 	VBox messages;
 
 	@FXML
 	ScrollPane scroller;
 
+	@FXML
+	TextArea textarea;
+
 	@Inject
 	ClientConfiguration clientConfiguration;
 
+	@Inject
+	private ContactRepository contactRepository;
+	@Inject
+	private DropMessageRepository dropMessageRepository;
+	@Inject
+	private Connector httpDropConnector;
+
+	Identity identity;
+	Contact c;
+	Date lastDate;
 
 	public void initialize(URL location, ResourceBundle resources) {
+
+		identity = clientConfiguration.getSelectedIdentity();
+		c = new Contact(identity, identity.getAlias(), identity.getDropUrls(), identity.getEcPublicKey());
 		try {
-			loadMessages();
+			loadMessages(c);
 		} catch (EntityNotFoundExcepion entityNotFoundExcepion) {
 			entityNotFoundExcepion.printStackTrace();
 		}
+
+		scroller.setVvalue(scroller.getVmax());
+		addListener();
 	}
 
-	void loadMessages() throws EntityNotFoundExcepion {
-		messages.getChildren().clear();
-		createMessageItems();
-		if(scroller.getVmax() == scroller.getVvalue()) {
-			scroller.setVvalue(scroller.getVmax());
+	private void addListener() {
+		((Region) scroller.getContent()).heightProperty().addListener((ov, old_val, new_val) -> {
+			if (scroller.getVvalue() != scroller.getVmax()) {
+				scroller.setVvalue(scroller.getVmax());
+			}
+		});
+		textarea.setOnKeyPressed(keyEvent -> {
+			if (keyEvent.getCode().equals(KeyCode.ENTER) && keyEvent.isControlDown()) {
+
+				try {
```

Comment:
this is the same code AND the same semantic as "handleSubmitButtonAction" (=>code duplication)